resources/frontend/js/convertkit.js

```diff
@@ -181,13 +181,24 @@ function convertStoreSubscriberEmailAsIDInCookie( emailAddress ) {
 			if ( convertkit.debug ) {
 				console.log( response );
 			}
+
+			return response.json();
 		}
 	)
 	.then(
 		function ( result ) {
 			if ( convertkit.debug ) {
 				console.log( result );
 			}
+
+			// Emit custom event with subscriber ID.
+			convertKitEmitCustomEvent(
```

Comment:
@n7studios Just a quick question, I noticed in another call, response sometimes returns `success: false`, if the user is not found (my case might have some edge cases because I was only on trial account with temp email, and email field would not render, so I kinda "hack" it by editing DOM to include the one from your platform by copying `email` field markup). This will only run once subscription is successful, correct? Thanks

Reply:
This isn't designed to be a guarantee. Edge cases will always exist - whether that be a race condition, confirmation email required etc.

If you need to truly perform an action once an email address subscribes, I'd recommend:
- registering a webhook in your solution: https://developers.convertkit.com/#create-a-webhook
- listening for the applicable subscribe event e.g. `subscriber.subscriber_activate`
- updating whatever existing event tracking you created using this emitter to confirm it truly happened

Thanks!